### PR TITLE
feat(api): make solution settings optional

### PR DESCRIPTION
**Pull request by jayzhang**
To support existing app feature, solution settings is now optional

### CODE REVIEW

**Review by LongOddCode — APPROVED** [review]

File: packages/fx-core/src/core/index.ts [hunk 1]
```diff
@@ -726,7 +733,14 @@ export class FxCore implements v3.ICore {
   ): Promise<Result<Void, FxError>> {
     currentStage = Stage.provision;
     inputs.stage = Stage.provision;
-    if (!ctx || !ctx.solutionV2 || !ctx.contextV2 || !ctx.envInfoV2) {
+    if (!ctx || !ctx.projectSettings) {
```
Comment (SLdragon) [inline packages/fx-core/src/core/index.ts:736]: how about simplify to `!ctx?.projectSettings`

Comment (jayzhang) [reply packages/fx-core/src/core/index.ts:736]: OK

**Review by KennethBWSong — APPROVED** [review]

**Review by SLdragon — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

File: packages/fx-core/src/core/index.ts [hunk 2]
```diff
@@ -265,52 +266,58 @@ export class FxCore implements v3.ICore {
...
-      );
-      if (createEnvResult.isErr()) {
-        return err(createEnvResult.error);
-      }
+      if (isCreatedFromExistingApp(inputs)) {
+        //TODO create from existing Tab or Bot/ME
+        // 1. call App Studio V3 API to create manifest with placeholder
+        // 2. create config.local.json to store existing App information
+      } else {
+        // there is no solution settings if created from existing app
```
Comment (dooriya) [inline packages/fx-core/src/core/index.ts:277]: Looks like this comment is for the `if` case, which is created from existing app.

Comment (jayzhang) [reply packages/fx-core/src/core/index.ts:277]: Yes, will be moved later

**Review by dooriya — APPROVED** [review]

**Pull request merged**